### PR TITLE
Small Spider Balance Changes and Tarantula QoL

### PR DESCRIPTION
**Pull request by IndieanaJones**
## About The Pull Request

This PR tweaks spiders a little bit in order to make them fairer to play as and against.  The changes are as follows:

- Broodmother and Nurse spiders can now create Sealed Web by doing the place web action on a tile with web already on it.  Sealed webs block atmos like a wall, but are just as fragile as normal webs.  They also block anything from passing through, including other spiders, so use them sparingly.

- Spiders now take 25% more burn damage.  This is mostly for flavor, but it should also allow lasers to be slightly more efficient at killing spiders.

- Spiders take 2 fewer atmos damage when in unsuitable temperatures (10 to 8).  This slightly compensates for the above change.

-  Broodmothers now take 15 seconds to lay eggs as opposed to 5 seconds.

- Broodmothers spawned by the spider infestation event or midround ruleset now spawn together as opposed to being spawned in two separate locations.  This was done as to have the spider horde flow out of one area as opposed to two, unless a little more elbow work is done by the spiders themselves.

- Tarantulas can now charge using right-click.  Removed the charge button ability.

## Why It's Good For The Game

Spiders can be extremely oppressive if the crew doesn't utilize atmos against them, but at the same time have no answer to atmos and are easily defeated if the crew uses atmos against them.  These changes try to level both extremes down some for more enjoyable middle ground gameplay.

## Changelog
:cl:
expansion: Spider nurses and broodmothers can now place down sealed webs by webbing an already webbed tile, allowing them to make weak walls.
qol: Tarantulas now use right-click to charge as opposed to a button ability.  Said button ability is now removed.
balance: Spiders have received some slight balance changes to make them less oppressive when they have the upper hand whilst letting them slightly handle atmos changes.  One major change is that broodmothers now take 15 seconds to lay eggs as opposed to 10 seconds.
qol: Spiders can see further in the dark now.
/:cl:

### CODE REVIEW

**Comment by wraith-54321** [comment]
I love it, before you could sneeze and kill spiders.

**Comment by Mothblocks** [comment]
I'm confused, this seems like mostly just a buff--the problem is that when you kill one spider, 3 more are ready to take their place. I don't know if 5 -> 15 is enough of a big deal to stop that, and it feels like in exchange they get more immunities?

**Comment by IndieanaJones** [comment]
I see this as mostly a nerf, so perhaps I should explain my thoughts on each change:

- first change is definitely a buff, as it prevents things like pouring acid on the floor to kill a whole nest.  Spiders can't walk through it though, and it's very fragile, so it's only so useful

- 2nd change is 100% a nerf

- Atmos damage is offset by the additional burn damage, 8 × 1.25 is 10.

- 100% a nerf

- Double-sided change

I'm typing this on phone so I'll edit this when I'm home, but perhaps this helps put things into perspective.

**Comment by IndieanaJones** [comment]
Well this is unfortunate.

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 1]
```diff
@@ -45,6 +44,7 @@
 	unique_name = 1
 	gold_core_spawnable = HOSTILE_SPAWN
 	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_VISIBLE
+	see_in_dark = 8
```
Comment (Kylerace) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:47]: this isnt documented or put into the changelog, why change this from 4 to 8?

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 2]
```diff
@@ -193,10 +196,11 @@
 	///The spider's charge ability
 	var/obj/effect/proc_holder/tarantula_charge/charge
```
Comment (Kylerace) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:197]: do you still have a need for this?

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 3]
```diff
@@ -502,7 +465,7 @@
 		spider.is_busy = TRUE
 		spider.visible_message(span_notice("[spider] begins to lay a cluster of eggs."),span_notice("You begin to lay a cluster of eggs."))
 		spider.stop_automated_movement = TRUE
-		if(do_after(spider, 100, target = get_turf(spider)))
```
Comment (Kylerace) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:505]: this isnt 5 seconds like what is said in the pr body this is 10

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 4]
```diff
@@ -256,11 +260,14 @@
...
 	poison_per_bite = 3
 	gold_core_spawnable = NO_SPAWN
+	web_sealer = TRUE
 	menu_description = "Royal spider variant specializing in reproduction and leadership, but has very low amount of health and deals low damage. Toxin injection of 3u per bite."
 	///If the spider is trying to cocoon something, what that something is.
 	var/atom/movable/cocoon_target
 	///How many humans this spider has drained but not layed enriched eggs for.
 	var/fed = 0
+	///How long it takes for a broodmother to lay eggs.
+	var/egg_lay_time = 15 SECONDS
```
Comment (Kylerace) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:270]: this should be in the changelog

**Review by Kylerace — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 1 above)
Comment (IndieanaJones) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm:47]: Ah, I forgot about this.  Honestly, this change is more of a convenience than anything else, limited vision in the dark isn't something considered in balancing the spiders, and this vision AOE applies extremely inconsistently.

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 3 above)
Comment (IndieanaJones) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm:505]: Are you sure?  I thought this timer goes down by 20 a second.

**Comment by moocowswag** [comment]
I dont think forcing broodmothers to sit still for 15 seconds at a time makes for interesting gameplay, especially when broodmothers are already at the mercy of there being ghosts who are even willing to become spiders.

But atleast 2 broods spawning together will encourage them to "have fun" and try to hunt a few crew members or die trying.

File: code/game/objects/effects/spiders.dm [hunk 5]
```diff
@@ -31,7 +31,10 @@
 	take_damage(5, BURN, 0, 0)
 
 /obj/structure/spider/stickyweb
+	///Whether or not the web is from the genetics power
 	var/genetic = FALSE
+	///Whether or not the web is a sealed web
+	var/sealed = FALSE
 	icon_state = "stickyweb1"
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/effects/spiders.dm:38]: In the future please put new variables below existing ones (I know genetics is also an offender but double cringe)

**Pull request merged**